### PR TITLE
Update to new conway-polynomials python package

### LINKED ISSUE #32747: Upstream Python distribution package for conway_polynomials
<div id="comment:0"></div>

... so that we can replace the Sage-specific installation script `build/pkgs/conway_polynomials/spkg-install.py` by just `pip install ...`

CC:  @orlitzky @antonio-rojas

Component: **packages: standard**

_Issue created by migration from https://trac.sagemath.org/ticket/32747_

### PR DESCRIPTION
**Pull request by orlitzky**
Fix https://github.com/sagemath/sage/issues/32747 by switching to the newly-minted [conway-polynomials package](https://pypi.org/project/conway-polynomials/0.7/) on pypi.

IMO `sage.databases.conway` (which makes the dict immutable by wrapping it in a class) is of dubious value but I've left everything alone for now.

### CODE REVIEW

File: build/pkgs/conway_polynomials/SPKG.rst [hunk 1]
```diff
@@ -1,12 +1,26 @@
-conway_polynomials: Tables of Conway polynomials over finite fields
+Python interface to Frank Lübeck's Conway polynomial database
```
Comment (mkoeppe) [inline build/pkgs/conway_polynomials/SPKG.rst]: Need to keep `conway_polynomials:` at the beginning - that's the general format (for the TOC in the package list)

File: src/sage/databases/conway.py [hunk 2]
```diff
@@ -152,7 +144,7 @@ def __len__(self):
 
             sage: c = ConwayPolynomials()
             sage: len(c)
-            35352
+            35357
         """
```
Comment (mkoeppe) [inline src/sage/databases/conway.py:146]: Oh? What happened there?

File: build/pkgs/conway_polynomials/SPKG.rst (hunk 1 above)
Comment (mkoeppe) [reply build/pkgs/conway_polynomials/SPKG.rst]: By the way, `./sage -package create conway_polynomials --pypi --type standard` sets up these things automatically in the right format

Comment (mkoeppe) [reply build/pkgs/conway_polynomials/SPKG.rst]: The `dependencies` file also needs updating

File: src/sage/databases/conway.py (hunk 2 above)
Comment (kiwifb) [reply src/sage/databases/conway.py:146]: I was asking myself the same question.

File: build/pkgs/conway_polynomials/SPKG.rst (hunk 1 above)
Comment (orlitzky) [reply build/pkgs/conway_polynomials/SPKG.rst]: Both should be fixed.

File: src/sage/databases/conway.py (hunk 2 above)
Comment (orlitzky) [reply src/sage/databases/conway.py:146]: Inflation.

The old package came with preprocessed data that would have been harder to diff against the upstream list for changes. I think it was just a tiny bit out-of-date. Here's the diff:

```diff
103a104
> [2,110,[1,1,1,0,0,0,1,0,0,0,1,1,0,1,1,0,0,1,0,0,1,0,0,0,0,1,1,0,1,1,0,0,0,1,0,1,1,1,0,1,0,0,1,1,0,0,0,0,1,0,0,0,0,0,1,1,1,1,1,1,0,0,1,0,1,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1]],
219a221
> [3,52,[2,0,1,1,1,2,0,1,2,0,0,0,1,0,2,0,2,2,2,1,2,0,2,1,0,0,2,2,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1]],
222a225,226
> [3,56,[2,2,0,0,0,1,0,2,0,0,0,0,2,1,1,0,0,1,2,2,2,2,0,2,2,1,2,2,0,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1]],
> [3,57,[1,1,2,2,2,2,1,0,1,1,0,2,1,0,2,2,2,1,0,2,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1]],
229a234
> [3,72,[2,2,0,2,1,1,1,2,1,0,2,1,2,1,0,1,0,2,1,1,0,2,1,2,1,2,0,2,0,1,1,1,2,2,1,0,2,2,1,1,0,0,1,2,0,1,1,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1]],
```

Comment (kiwifb) [reply src/sage/databases/conway.py:146]: I see. Does upstream update the list "often"?

File: build/pkgs/conway_polynomials/SPKG.rst [hunk 3]
```diff
@@ -1,4 +1,4 @@
-Python interface to Frank Lübeck's Conway polynomial database
+conway_polynomials: Python interface to Frank Lübeck's Conway polynomial database
 ===================================================================
```
Comment (mkoeppe) [inline build/pkgs/conway_polynomials/SPKG.rst]:
```suggestion
=================================================================================
```

Comment (orlitzky) [reply build/pkgs/conway_polynomials/SPKG.rst]: argh

File: src/sage/databases/conway.py (hunk 2 above)
Comment (orlitzky) [reply src/sage/databases/conway.py:146]: No idea. There's no change history on the site, but the new package ships the upstream file unmodified so at least we can `git diff` it in the future.

Comment (kiwifb) [reply src/sage/databases/conway.py:146]: Shipping upstream file unmodified sounds very good to me. I think it is ready.

**Comment by dimpase** [comment]
isn't this duplicating data available from GAP ?

**Comment by mkoeppe** [comment]
We discussed that in #32747.

**Comment by mkoeppe** [comment]
```
sage -t --long --random-seed=286735480429121101562228604801325644303 sage/databases/all.py  # 1 doctest failed
sage -t --long --random-seed=286735480429121101562228604801325644303 sage/features/__init__.py  # 1 doctest failed
```

**Review by mkoeppe — APPROVED** [review]
LGTM, thanks for working on this!

**Comment by orlitzky** [comment]
No problem, thanks for the help. I'll work on the others as time permits. We're in the single-digits now for missing Gentoo packages and zero is starting to look possible.

File: src/sage/features/databases.py [hunk 4]
```diff
@@ -44,16 +44,13 @@ def __init__(self):
...
-            search_path = [CONWAY_POLYNOMIALS_DATA_DIR]
-        else:
-            search_path = []
-        StaticFile.__init__(self, "conway_polynomials",
-                            filename='conway_polynomials.p',
-                            search_path=search_path,
-                            spkg='conway_polynomials',
-                            description="Frank Luebeck's database of Conway polynomials",
-                            type='standard')
+        PythonModule.__init__(
```
Comment (tobiasdiez) [inline src/sage/features/databases.py]:
```suggestion
        super().__init__(
```
(not that it is that important)

**Review by tobiasdiez — CHANGES_REQUESTED** [review]
Nice, thanks for working on this.

The conda tests are failing, e.g. because of
```
File "src/sage/combinat/designs/database.py", line 820, in sage.combinat.designs.database.OA_9_135
Failed example:
    OA = OA_9_135()                                                           # needs sage.rings.finite_rings sage.schemes
Exception raised:
    Traceback (most recent call last):
      File "/home/runner/work/sage/sage/src/sage/doctest/forker.py", line 709, in _run
        self.compile_and_execute(example, compiler, test.globs)
      File "/home/runner/work/sage/sage/src/sage/doctest/forker.py", line 1144, in compile_and_execute
        exec(compiled, globs)
      File "<doctest sage.combinat.designs.database.OA_9_135[2]>", line 1, in <module>
        OA = OA_9_135()                                                           # needs sage.rings.finite_rings sage.schemes
      File "/home/runner/work/sage/sage/src/sage/combinat/designs/database.py", line 837, in OA_9_135
        G,B = singer_difference_set(16,2)
      File "/home/runner/work/sage/sage/src/sage/combinat/designs/difference_family.py", line 390, in singer_difference_set
        c = conway_polynomial(p,e*(d+1))
      File "/home/runner/work/sage/sage/src/sage/rings/finite_rings/conway_polynomials.py", line 63, in conway_polynomial
        return R(ConwayPolynomials()[p][n])
      File "/home/runner/work/sage/sage/src/sage/databases/conway.py", line 96, in __init__
        import conway_polynomials
    ModuleNotFoundError: No module named 'conway_polynomials'
```

So we need to also create a conda feedstock for it.

I think we can also remove the conway_polynomials feature (these failures show that it's presence or absence doesn't really imply anything).

**Comment by mkoeppe** [comment]
Right, conway-polynomials needs to be added to "install-requires" in src/setup.cfg.m4

Adding conda packaging is a downstream activity and not the burden of the PR author

**Comment by tobiasdiez** [comment]
> Adding conda packaging is a downstream activity and not the burden of the PR author

How is that a downstream activity if we are the ones offering the package?!
I've now opened https://github.com/conda-forge/staged-recipes/pull/24585 for this, does anyone of you wants to be added as maintainer? Would be nice if a license file could be added as well (https://github.com/sagemath/conway-polynomials/issues/1).

**Comment by mkoeppe** [comment]
> > Adding conda packaging is a downstream activity and not the burden of the PR author
> 
> How is that a downstream activity if we are the ones offering the package?!

Upstream = offers source repo and PyPI package

Downstream = provides Debian, Fedora, OpenSuSE, Arch Linux, Conda, Gentoo, ... packaging.

File: src/sage/features/databases.py [hunk 5]
```diff
@@ -19,43 +19,10 @@
 
 from . import StaticFile, PythonModule
 from sage.env import (
-    CONWAY_POLYNOMIALS_DATA_DIR,
     CREMONA_MINI_DATA_DIR, CREMONA_LARGE_DATA_DIR,
     POLYTOPE_DATA_DIR)
 
 
-class DatabaseConwayPolynomials(StaticFile):
```
Comment (mkoeppe) [inline src/sage/features/databases.py:27]: Normally I think we would say that this is API and should not be removed without deprecation. But I don't object to removing it

**Comment by tobiasdiez** [comment]
It's now available on [conda-forge](https://github.com/conda-forge/conway-polynomials-feedstock), so it should work now once you added a `conda.txt` file.

**Review by tobiasdiez — APPROVED** [review]
LGTM

**Comment by orlitzky** [comment]
We should update this to v0.8 first but my sage has been occupied all day running gap tests.

**Pull request merged**